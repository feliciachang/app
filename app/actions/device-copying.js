import _ from 'lodash';

import * as Types from './types';

export function copyFromDevices(devices) {
    return {
        type: Types.COPY_DEVICE_FILES,
        devices: devices
    };
}

