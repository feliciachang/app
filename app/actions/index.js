'use strict';

import * as ActionTypes from './types';

export * from './navigation';
export * from './emails';
export * from './device-status';
export * from './device-data';
export * from './timers';
export * from './modules';
export * from './configuration';
export * from './live-data';

export const Types = ActionTypes;
