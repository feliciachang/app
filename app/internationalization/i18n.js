import RNLanguages from 'react-native-languages';
import i18n from 'i18n-js';

import en from './translations/en.json';
import es from './translations/es.json';
import pt from './translations/pt.json';
import fr from './translations/fr.json';

i18n.locale = RNLanguages.language;

i18n.fallbacks = true;
i18n.translations = { pt, en, es, fr};

export default i18n;
