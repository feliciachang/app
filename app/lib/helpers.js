'use strict';

export function unixNow() {
    return Math.round((new Date()).getTime() / 1000);
}
