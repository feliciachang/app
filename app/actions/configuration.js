import { CALL_DEVICE_API } from '../middleware/device-api';
import { QueryType } from '../lib/protocol';

import * as Types from './types';

export function queryConfiguration() {
    return {
        [CALL_DEVICE_API]: {
            types: [Types.DEVICE_NETWORK_CONFIGURATION_START, Types.DEVICE_NETWORK_CONFIGURATION_SUCCESS, Types.DEVICE_NETWORK_CONFIGURATION_FAIL],
            blocking: true,
            message: {
                type: QueryType.values.QUERY_NETWORK_SETTINGS,
            }
        }
    };
}

export function saveNetworkConfiguration(newConfiguration) {
    return {
        [CALL_DEVICE_API]: {
            types: [Types.DEVICE_SAVE_NETWORK_CONFIGURATION_START, Types.DEVICE_SAVE_NETWORK_CONFIGURATION_SUCCESS, Types.DEVICE_SAVE_NETWORK_CONFIGURATION_FAIL],
            blocking: true,
            message: {
                type: QueryType.values.QUERY_CONFIGURE_NETWORK_SETTINGS,
                networkSettings: newConfiguration,
            }
        }
    };
}

