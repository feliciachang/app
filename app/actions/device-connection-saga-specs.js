import 'react-native';

import SagaTester from 'redux-saga-tester';
import Promise from "bluebird";

import * as Types from './types';
import { findDeviceInfo } from './discovery';
import { discoverDevices, pingConnectedDevice } from './sagas';
import { useFakeDeviceConnection } from '../middleware/device-api';
import { QueryType, ReplyType } from '../lib/protocol';

describe('device connection saga', () => {
    let tester;
    let fakeDevice;

    beforeEach(() => {
        fakeDevice = useFakeDeviceConnection();
        tester = new SagaTester({
            initialState: {
                devices: { },
                deviceStatus: {
                    addresses: { },
                    connected: null,
                    api: {
                        pending: false
                    }
                }
            }
        });
    });

    describe('discovery', () => {
        let task;

        beforeEach(() => {
            task = tester.start(discoverDevices);
        });

        it.skip('should timeout and fail after timeout interval', async () => {
            await tester.waitFor(Types.FIND_DEVICE_FAIL);
        });

        it.skip('should timeout and fail when discovery returns invalid addresses', async () => {
            tester.dispatch(findDeviceInfo('127.0.0.1', 0));
            await tester.waitFor(Types.FIND_DEVICE_FAIL);
        });

        it('should query the device for capabilities and return success', async () => {
            fakeDevice.push({}, {
                type: ReplyType.values.REPLY_CAPABILITIES,
                response: { capabilities: { } }
            })
            tester.dispatch(findDeviceInfo('127.0.0.1', 12345));
            await tester.waitFor(Types.FIND_DEVICE_SUCCESS);
            expect(fakeDevice.queue).toHaveLength(0);
        });

        afterEach(() => {
            task.cancel();
        });
    });

    describe('pinging', () => {
        let task;

        beforeEach(() => {
            task = tester.start(pingConnectedDevice);
        });

        afterEach(() => {
            // Cancel to keep from interfering with the following test.
            task.cancel();
        });

        it('should ping after selecting device', () => {
            tester.getState().deviceStatus.connected = {};

            fakeDevice.push({}, {
                type: ReplyType.values.REPLY_CAPABILITIES,
                response: { capabilities: { } }
            });

            tester.dispatch({
                type: Types.FIND_DEVICE_SELECT,
                address: {}
            });

            return Promise.delay(Config.pingDeviceInterval + 100).then(() => {
                expect(fakeDevice.queue).toHaveLength(0);
            });
        });

        it('should ping after a previous ping', () => {
            tester.getState().deviceStatus.connected = {};

            fakeDevice.push({}, {
                type: ReplyType.values.REPLY_CAPABILITIES,
                response: { capabilities: { } }
            });

            tester.dispatch({
                type: Types.DEVICE_PING_SUCCESS,
                address: {}
            });

            return Promise.delay(Config.pingDeviceInterval + 100).then(() => {
                expect(fakeDevice.queue).toHaveLength(0);
            });
        });

        it('should dispatch FIND_DEVICE_LOST when ping fails', async () => {
            tester.getState().deviceStatus.connected = {};

            fakeDevice.pushError({}, "No reply");

            tester.dispatch({
                type: Types.FIND_DEVICE_SELECT,
                address: {}
            });

            await tester.waitFor(Types.FIND_DEVICE_LOST);
        });
    });
});
