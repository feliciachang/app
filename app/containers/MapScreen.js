import React from "react";
import PropTypes from "prop-types";
import { createStore } from "redux";
import { connect, Provider } from "react-redux";
import _ from "lodash";

import RNLanguages from "react-native-languages";
import i18n from "../internationalization/i18n";

import Config from "../config";

import { View, Text, Image } from "react-native";

import { AppScreen, MenuButtonContainer, MenuButton } from "../components";
import { queryFiles } from "../actions";

import { navigateBack, navigateMap, navigateWelcome, location } from "../actions";

import { StyleSheet } from "react-native";

import Mapbox from "@mapbox/react-native-mapbox-gl";
import { MAPBOX_ACCESS_TOKEN, MAPBOX_STYLE } from "../../secrets";

const styles = StyleSheet.create({
    container: {
        display: "flex",
        flex: 1
    },
    annotationContainer: {
        width: 30,
        height: 30,
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "white",
        borderRadius: 15
    },
    annotationFill: {
        width: 30,
        height: 30,
        borderRadius: 15,
        backgroundColor: "orange",
        transform: [{ scale: 0.6 }]
    }
});

Mapbox.setAccessToken(MAPBOX_ACCESS_TOKEN);

class MapScreen extends React.Component {
    static navigationOptions = ({ navigation }) => {
        return { title: i18n.t("mapScreen.title") };
    };

    componentDidMount() {
        this.props.location();
    }

    renderAnnotations(coordinate, i) {
        const myString = coordinate.toString();
        return (
            <Mapbox.PointAnnotation key={i} id={"pointAnnotation-" + i} coordinate={coordinate}>
                <View style={styles.annotationContainer}>
                    <View style={styles.annotationFill} />
                </View>
                <Mapbox.Callout title={myString} />
            </Mapbox.PointAnnotation>
        );
    }

    render() {
        const { navigateBack, navigateMap, navigateWelcome, giveLocation, records } = this.props;
        if (giveLocation.phone !== undefined) {
            const coordinateArray = [giveLocation.phone.long, giveLocation.phone.lat];
            return (
                <View style={styles.container}>
                    <Mapbox.MapView styleURL={MAPBOX_STYLE} zoomLevel={15} centerCoordinate={coordinateArray} style={styles.container}>
                        <Mapbox.ShapeSource id="line1" shape={giveLocation.route}>
                            <Mapbox.LineLayer id="linelayer1" style={{ lineColor: "red" }} />
                        </Mapbox.ShapeSource>
                        {giveLocation.sensors.map((coordinate, index) => {
                            return this.renderAnnotations(coordinate, index);
                        })}
                        {this.renderAnnotations(coordinateArray, 3)}
                    </Mapbox.MapView>
                </View>
            );
        } else {
            return (
                <View style={styles.container}>
                    <Text>loading map</Text>
                </View>
            );
        }
    }
}

MapScreen.propTypes = {
    navigateWelcome: PropTypes.func.isRequired,
    navigateBack: PropTypes.func.isRequired
};

const mapStateToProps = state => {
    //const route = state.nav.routes[state.nav.index];
    //const path = route.params.path;
    return {
        //path: path,
        giveLocation: state.giveLocation
        //records: state.localFiles.records[path] || {}
    };
};

export default connect(
    mapStateToProps,
    {
        navigateMap,
        location,
        navigateBack,
        navigateWelcome,
        queryFiles
    }
)(MapScreen);
