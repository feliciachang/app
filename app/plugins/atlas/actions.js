import { delay } from 'redux-saga';
import { put, take, takeLatest, takeEvery, select, all, race, call } from 'redux-saga/effects';

import * as AppActionTypes from '../../actions/types';
import { QueryType as AppQueryType } from '../../lib/protocol';

import * as AtlasActionTypes from './types';
import { deviceCall } from '../../actions/saga-utils';
import { CALL_DEVICE_API } from '../../middleware/device-api';

import { QueryType, SensorType, atlasSensorQuery, encodeWireAtlasQuery, decodeWireAtlasReply } from './protocol';

export function atlasCalibrationBegin(sensor) {
    return {
        type: AtlasActionTypes.ATLAS_CALIBRATION_BEGIN,
        sensor: sensor,
    };
}

export function atlasCalibrationEnd() {
    return {
        type: AtlasActionTypes.ATLAS_CALIBRATION_END,
    };
}

export function atlasReadSensor(sensor) {
    return {
        type: AtlasActionTypes.ATLAS_READ_SENSOR_START,
        sensor: sensor,
    };
}

export function atlasSensorCommand(types, blocking, sensor, command) {
    const query = atlasSensorQuery(sensor, command);
    const encoded = encodeWireAtlasQuery(query);

    return (dispatch, getState) => {
        return dispatch({
            [CALL_DEVICE_API]: {
                types: types,
                address: getState().deviceStatus.connected,
                blocking: blocking,
                message: {
                    type: AppQueryType.values.QUERY_MODULE,
                    module: {
                        id: 8,
                        address: 8,
                        message: encoded
                    }
                }
            },
        });
    };
}

export function atlasSetProbeType(sensor, command) {
    const types = [AtlasActionTypes.DEVICE_ATLAS_SENSOR_SET_PROBE_TYPE_START, AtlasActionTypes.DEVICE_ATLAS_SENSOR_SET_PROBE_TYPE_SUCCESS, AtlasActionTypes.DEVICE_ATLAS_SENSOR_SET_PROBE_TYPE_FAIL];
    return atlasSensorCommand(types, true, sensor, command);
}

export function atlasCalibrate(sensor, command) {
    const types = [AtlasActionTypes.DEVICE_ATLAS_SENSOR_CALIBRATE_START, AtlasActionTypes.DEVICE_ATLAS_SENSOR_CALIBRATE_SUCCESS, AtlasActionTypes.DEVICE_ATLAS_SENSOR_CALIBRATE_FAIL];
    return atlasSensorCommand(types, true, sensor, command);
}

export function atlasReading(sensor) {
    const types = [AtlasActionTypes.DEVICE_ATLAS_SENSOR_READING_START, AtlasActionTypes.DEVICE_ATLAS_SENSOR_READING_SUCCESS, AtlasActionTypes.DEVICE_ATLAS_SENSOR_READING_FAIL];
    return atlasSensorCommand(types, false, sensor, "R");
}

function* takeAtlasReadings() {
    yield takeLatest([AtlasActionTypes.ATLAS_READ_SENSOR_START], function* (start) {
        while (true) {
            yield call(deviceCall, atlasReading(start.sensor));

            const { stopped, to } = yield race({
                stopped: take([AtlasActionTypes.ATLAS_READ_SENSOR_STOP, AtlasActionTypes.ATLAS_CALIBRATION_END]),
                to: delay(2000)
            });

            if (stopped) {
                break;
            }
        }
    });
}

export function* atlasSagas() {
    yield all([
        takeAtlasReadings()
    ]);
}
