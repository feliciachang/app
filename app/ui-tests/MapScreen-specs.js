import React from "react";
import MapScreen from "../containers/MapScreen";
import { mockFunctionHelper } from "../../specs/helpers";

import { shallow, mount, render } from "enzyme";
import i18n from "../internationalization/i18n";
import toJson from "enzyme-to-json";
import * as Types from "../actions/index";

let store = mockFunctionHelper({
  giveLocation: {
    phone: { type: "PHONE_LOC", lat: 34.0314141, long: -118.2663902 },
    sensors: [[-119, 36], [-120, 37], [-118, 34]]
  }
});
describe("<MapScreen/>", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(<MapScreen store={store} />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
