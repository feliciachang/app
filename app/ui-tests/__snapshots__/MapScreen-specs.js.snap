// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapScreen/> render() renders the component 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
    }
  }
>
  <MapView
    animated={false}
    attributionEnabled={true}
    centerCoordinate={
      Array [
        -118.2663902,
        34.0314141,
      ]
    }
    heading={0}
    localizeLabels={false}
    logoEnabled={true}
    pitch={0}
    pitchEnabled={true}
    regionDidChangeDebounceTime={500}
    regionWillChangeDebounceTime={10}
    rotateEnabled={true}
    scrollEnabled={true}
    style={
      Object {
        "display": "flex",
        "flex": 1,
      }
    }
    styleURL="mapbox://styles/mapbox/satellite-v9"
    surfaceView={false}
    userTrackingMode="None"
    zoomLevel={15}
  >
    <ShapeSource
      id="line1"
    >
      <LineLayer
        id="linelayer1"
        sourceID="DefaultSourceID"
        style={
          Object {
            "lineColor": "red",
          }
        }
      />
    </ShapeSource>
    <PointAnnotation
      anchor={
        Object {
          "x": 0.5,
          "y": 0.5,
        }
      }
      coordinate={
        Array [
          -119,
          36,
        ]
      }
      id="pointAnnotation-0"
      key="0"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": 15,
              "height": 30,
              "transform": Array [
                Object {
                  "scale": 0.6,
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
      <Callout
        title="-119,36"
      />
    </PointAnnotation>
    <PointAnnotation
      anchor={
        Object {
          "x": 0.5,
          "y": 0.5,
        }
      }
      coordinate={
        Array [
          -120,
          37,
        ]
      }
      id="pointAnnotation-1"
      key="1"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": 15,
              "height": 30,
              "transform": Array [
                Object {
                  "scale": 0.6,
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
      <Callout
        title="-120,37"
      />
    </PointAnnotation>
    <PointAnnotation
      anchor={
        Object {
          "x": 0.5,
          "y": 0.5,
        }
      }
      coordinate={
        Array [
          -118,
          34,
        ]
      }
      id="pointAnnotation-2"
      key="2"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": 15,
              "height": 30,
              "transform": Array [
                Object {
                  "scale": 0.6,
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
      <Callout
        title="-118,34"
      />
    </PointAnnotation>
    <PointAnnotation
      anchor={
        Object {
          "x": 0.5,
          "y": 0.5,
        }
      }
      coordinate={
        Array [
          -118.2663902,
          34.0314141,
        ]
      }
      id="pointAnnotation-3"
      key="3"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": 15,
              "height": 30,
              "transform": Array [
                Object {
                  "scale": 0.6,
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
      <Callout
        title="-118.2663902,34.0314141"
      />
    </PointAnnotation>
  </MapView>
</View>
`;
